### DISCUSSION TITLE
"could not find option 'match' in the list of available options" failure on beta lane

### DISCUSSION THREAD

**Discussion by iainsimons** [fastlane/fastlane#18171] [Q&A]
Hello there!

I'm hitting an error on attempting to run fastlane beta

```could not find option 'match' in the list of available options```

and can't identify the cause. I've attempted reinstalling gems, but am also bumping up against my scant knowledge of Ruby. 
The operation works fine without the match command.
Any help much appreciated.

Fastlane env: 
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.7                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.7 (19H114)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 12.4                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_GB.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    	match(type: "appstore")
 api_key = app_store_connect_api_key(
    key_id: "xxxxxx",
    issuer_id: "xxxxxxx,
    key_filepath: "/Users/builder/Downloads/certs/AuthKey_33GAG9BBHC.p8",
    duration: 1200, # optional
    in_house: false, # optional but may be required if using match/sigh
  )

	 increment_build_number(
    build_number: latest_testflight_build_number + 1,
    xcodeproj: "Unity-iPhone.xcodeproj"
  ) 


    build_app(workspace: "Unity-iPhone.xcworkspace", scheme: "Unity-iPhone",export_xcargs: "-allowProvisioningUpdates")

    upload_to_testflight(api_key: api_key)

  end
end

```
</details>

<details><summary>`./Appfile`</summary>

```ruby
app_identifier("com.xxxx.xxxx") # The bundle identifier of your app
apple_id("xxxx@xxx.xxx") # Your Apple email address

itc_team_id("xxxxx") # App Store Connect Team ID
team_id("xxxxx") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.174.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.3.0        |
| slack-notifier                | 2.3.2        |
| atomos                        | 0.1.3        |
| claide                        | 1.0.3        |
| colored2                      | 3.1.2        |
| nanaimo                       | 0.3.0        |
| xcodeproj                     | 1.19.0       |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| plist                         | 3.6.0        |
| public_suffix                 | 4.0.6        |
| addressable                   | 2.7.0        |
| multipart-post                | 2.0.0        |
| word_wrap                     | 1.0.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| artifactory                   | 3.0.15       |
| babosa                        | 1.0.4        |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| excon                         | 0.79.0       |
| faraday-net_http              | 1.0.1        |
| ruby2_keywords                | 0.0.4        |
| faraday                       | 1.3.0        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.2        |
| gh_inspector                  | 1.1.3        |
| json                          | 2.1.0        |
| mini_magick                   | 4.11.0       |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| xcpretty-travis-formatter     | 1.0.1        |
| dotenv                        | 2.7.6        |
| naturally                     | 2.2.1        |
| simctl                        | 1.6.8        |
| jwt                           | 2.2.2        |
| uber                          | 0.1.0        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| mini_mime                     | 1.0.2        |
| multi_json                    | 1.15.0       |
| signet                        | 0.14.1       |
| httpclient                    | 2.8.3        |
| google-api-client             | 0.38.0       |
| google-cloud-env              | 1.4.0        |
| google-cloud-errors           | 1.0.1        |
| google-cloud-core             | 1.5.0        |
| google-apis-core              | 0.2.1        |
| google-apis-iamcredentials_v1 | 0.1.0        |
| digest-crc                    | 0.6.3        |
| google-cloud-storage          | 1.30.0       |
| emoji_regex                   | 3.2.1        |
| jmespath                      | 1.4.0        |
| aws-eventstream               | 1.1.0        |
| aws-sigv4                     | 1.2.2        |
| aws-sdk-core                  | 3.112.0      |
| aws-sdk-kms                   | 1.42.0       |
| aws-sdk-s3                    | 1.88.0       |
| CFPropertyList                | 3.0.3        |
| bundler                       | 2.2.9        |
| os                            | 1.1.1        |
| memoist                       | 0.16.2       |
| googleauth                    | 0.15.1       |
| rexml                         | 3.2.4        |
| webrick                       | 1.7.0        |
| google-apis-storage_v1        | 0.2.0        |
| rake                          | 13.0.3       |
| aws-partitions                | 1.426.0      |
| forwardable                   | 1.2.0        |
| logger                        | 1.3.0        |
| date                          | 2.0.0        |
| stringio                      | 0.0.2        |
| zlib                          | 1.0.0        |
| ipaddr                        | 1.2.2        |
| openssl                       | 2.1.2        |
| ostruct                       | 0.1.0        |
| strscan                       | 1.0.0        |
| fileutils                     | 1.1.0        |
| etc                           | 1.0.1        |
| io-console                    | 0.4.7        |
| libxml-ruby                   | 3.1.0        |
| psych                         | 3.1.0        |
| mutex_m                       | 0.1.0        |
</details>

*generated on:* **2021-02-10**
</details>

**Comment by rzehan** [comment]
Had the same issue, on my side I was calling match() in Matchfile, so removing it solved this issue.
